### PR TITLE
feat(rules): Add rule to check for mutations of loop iterator

### PR DESCRIPTION
**Pull request by mimre25**
This implements a POC for #445.

I opted to go the route of alerting for any mutations of a loop iterator.

Note: 
I picked B999 temporarily, not sure how the numbers are picked.
Also, I'm very much open to change the error message.

Would be happy if you could give this a review @cooperlees  :slightly_smiling_face:

### CODE REVIEW

File: bugbear.py [hunk 1]
```diff
@@ -495,6 +496,7 @@ def visit_For(self, node):
         self.check_for_b020(node)
         self.check_for_b023(node)
         self.check_for_b031(node)
+        self.check_for_b999(node)
```
Comment (cooperlees) [inline bugbear.py:499]:
```suggestion
        self.check_for_b038(node)
```
We generally just go next number unless we make optional - They are >900.

File: bugbear.py [hunk 2]
```diff
@@ -1544,6 +1546,18 @@ def check(num_args, param_name):
         elif node.func.attr == "split":
             check(2, "maxsplit")
 
+    def check_for_b999(self, node: ast.For):
```
Comment (cooperlees) [inline bugbear.py:1549]:
```suggestion
    def check_for_b038(self, node: ast.For):
```

File: bugbear.py [hunk 3]
```diff
@@ -2045,6 +2102,8 @@ def visit_Lambda(self, node):
         " statement."
     )
 )
+
 B950 = Error(message="B950 line too long ({} > {} characters)")
 
+B999 = Error(message="B999 editing loop iterable can cause unintended bugs!")
```
Comment (cooperlees) [inline bugbear.py:2108]:
```suggestion
B999 = Error(message="B038 editing a loop's mutable iterable often leads to unexpected results/bugs")
```
I'm not tied to this, but just tried to be as explicit and descriptive as possible

**Review by cooperlees — CHANGES_REQUESTED** [review]
Many thanks for this. Looks good. I like this starting point.

Can you please:
- Lets go with b038
- Add more detailed few lines explanation into READNE.rst

File: bugbear.py (hunk 3 above)
Comment (mimre25) [reply bugbear.py]: Sounds good!

**Review by cooperlees — APPROVED** [review]
Many thanks for your contribution!

Will get a release our next week.

**Comment by sahuagin** [comment]
I have an example for a false positive (I hope), if it's desired.
Setup. A CSV file as input with the first item in each row being an index number.
```python
        id_pos = 0
        # lf is LogFile not LazyFrame
        lf = self._logger
        with self.rlock:
            lf.seek(0, 0)
            cpos = lf.tell()
            for line in lf:
                if len(line.strip()) == 0:
                    continue
                sl = line.split(b",", 1)
                self._mapper[sl[id_pos]] = cpos
                cpos = lf.tell()
```
Desired behavior is to find all indexes and record the file position where they start.
This just started getting flagged, which led me here. There are 4 more flags in the same file,
tell()
seek() x2
write()

**Comment by JelleZijlstra** [comment]
@sahuagin thanks for the report, #453 should fix this.

**Pull request merged**